code/modules/implant_hacking/board.dm

```diff
@@ -0,0 +1,288 @@
+			break
+
+		possible_directions = list()
+
+		counter++
+
+		xcord = queued_coord_list[counter][1]
+		ycord = queued_coord_list[counter][2]
+
+		var/list/maybe_dir = board[xcord][ycord].get_dir_val_list()
+
+		if(xcord > 1 && ((WEST in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += WEST
+
+		if(xcord < board_size && ((EAST in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += EAST
+
+		if(ycord > 1 && ((NORTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += NORTH
+
+		if(ycord < board_size && ((SOUTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += SOUTH
+
+		board[xcord][ycord].connected = FALSE
+
+		for(var/dir in possible_directions)
+
+			switch(dir)
+
+				if(NORTH)
+
+					dirlist = board[xcord][ycord-1].get_dir_val_list()
+
+					if(board[xcord][ycord-1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(SOUTH in dirlist)
+						if(!board[xcord][ycord-1].visited)
+							queued_coord_list += list(list(xcord,ycord-1))
+						board[xcord][ycord].connected = TRUE
+
+				if(SOUTH)
+
+					dirlist = board[xcord][ycord+1].get_dir_val_list()
+
+					if(board[xcord][ycord+1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(NORTH in dirlist)
+						if(!board[xcord][ycord+1].visited)
+							queued_coord_list += list(list(xcord,ycord+1))
+						board[xcord][ycord].connected = TRUE
+
+				if(WEST)
+
+					dirlist = board[xcord-1][ycord].get_dir_val_list()
+
+					if(board[xcord-1][ycord].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(EAST in dirlist)
+						if(!board[xcord-1][ycord].visited)
+							queued_coord_list += list(list(xcord-1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+				if(EAST)
+
+					dirlist = board[xcord+1][ycord].get_dir_val_list()
+
+					if(board[xcord+1][ycord].finish)
+						return TRUE
+
+					if(WEST in dirlist)
+						if(!board[xcord+1][ycord].visited)
+							queued_coord_list += list(list(xcord+1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+		board[xcord][ycord].visited = TRUE
+
+	return FALSE
+/**
+ * Simple holder of information and some procs relating to this information
+ *
+ * Holds relevant information about a single cell of the board.
+ */
+/datum/hacking_minigame_piece
+	var/rotation = 0
+	var/pass_in = NONE
+	var/pass_out = NONE
+	var/visited = FALSE
+	var/start = FALSE
+	var/finish = FALSE
+	var/connected = FALSE
+
+	var/datum/hacking_minigame/game
+
+/datum/hacking_minigame_piece/New(datum/hacking_minigame/_game)
+	. = ..()
+	game = _game
+
+///Returns value of pass_in + pass_out adjusted for rotation
+/datum/hacking_minigame_piece/proc/get_dir_val()
+	if(pass_in != NONE && pass_out != NONE)
+		return angle2dir(dir2angle(pass_in)+rotation*90) + angle2dir(dir2angle(pass_out)+rotation*90)
```

Comment:
because then you don't need to do this

Reply:
I *really* dislike how byond handles directions, and working with angles is much much simpler than having to constantly have a spreadsheet open of what arbitrary value means in terms of degrees.

Reply:
except you're not working with degrees you're working with 4 arbitrary states of direction, bitshifting will prevent you from having this convoluted function

Reply:
god i fucking hate this, i hate it that the directions dont go clockwise or counter clockwise but instead go in this illogical up down right left motion, who made this, and who hurt them.

Reply:
i reworked the procs using turn in rotate()